stag-library-compiler/src/main/java/com/vimeo/stag/processor/utils/ElementUtils.java

```diff
@@ -72,6 +76,42 @@ public static String getPackage(@NotNull TypeMirror type) {
     }
 
     /**
+     * Get the annotation applied of the specified type if it has been applied to the provided
+     * element.  This method will check the supplied element and work up the type hierarchy,
+     * returning the first annotation instance found.
+     *
+     * @param annotationType annotation type to search for
+     * @param element element to analyze
+     * @param <T> annotation type
+     * @return annotation instance or {@code null} if not found
+     */
+    @Nullable
+    public static <T extends Annotation> T findAnnotation(
+            @NotNull Class<T> annotationType, @NotNull Element element) {
+        T annotatedElement = element.getAnnotation(annotationType);
+        if (annotatedElement != null)
+            return annotatedElement;
+
+        if (element instanceof TypeElement) {
+            TypeElement typeElement = (TypeElement) element;
+            TypeMirror superTypeMirror = typeElement.getSuperclass();
+            if (superTypeMirror != null) {
+                Element superTypeElement = TypeUtils.getElementFromTypeMirror(superTypeMirror);
+                return superTypeElement == null ? null : findAnnotation(annotationType, superTypeElement);
+            }
+        }
```

Comment:
Okay, I think this is still valid - at least until you merge the removal of `@GsonAdapterKey`.  The code in question is still needed to walk up the inheritance hierarchy to determine what field options are set.  Do I read that correctly?